Bot Sharder/sharded-bot.js

```diff
@@ -1,172 +1,97 @@
-		  object[prop[1]] = value[1] ;
-		  return object
-		}
-	  })
-	  .reduce((obj, item) => {
-		let prop = Object.keys(item)[0];
-		obj[prop] = item[prop];
-		return obj;
-	  }, {});
-  
-	
-	if(args && args.shard_count){
-		console.log(`Command Line Arg: shard_count=${args.shard_count}`)
-		totalShards = args.shard_count;
-	}
-
-
-console.log(y(`Starting the DBM Bot with ${totalShards} total shards...`))
-
-
-function getToken(){
-	// dbm's encryption system
-	var crypto = require('crypto');
-
-	let password ='';
-
-	try {
-		password = require('discord-bot-maker');
-	} catch(e) {}
-
-	var decrypt = function(text) {
-		if(password.length === 0) return text;
-		const decipher = this.crypto.createDecipher('aes-128-ofb', password);
-		let dec = decipher.update(text, 'hex', 'utf8');
-		dec += decipher.final('utf8');
-		return dec;
-	};
-
-	const fs = require('fs');
-	const path = require('path');	
-
-	const filePath = path.join(process.cwd(),'data', 'settings.json');
-
-
-	if(fs.existsSync(filePath)){
-		var content = fs.readFileSync(filePath) 
-		try {
-			if(typeof content !== 'string' && content.toString) content = content.toString();
-			return JSON.parse(decrypt(content)).token;
-		} catch(e) {
-			console.error(`There was issue parsing settings.json!`);
-			return;
-		}
-
-	}else{
-		console.error(`settings.json does not exist!`);
-	}					
+// Made by TheMonDon#1721
+// Some code by General Wrex
+const version = '1.0.1';
+
+// Include discord.js and original check
+const { version: _version, ShardingManager } = require('discord.js');
+if (_version < '12.0.0') {
+  console.log('This version of Discord Bot Maker requires Discord.JS v12.\nPlease use "Project > Module Manager" and "Project > Reinstall Node Modules" to update to Discord.JS v12.');
+  throw new Error('Need Discord.JS v12 to Run!!!');
 }
 
-if(!getToken()){	
-	console.error(r(`Token must be supplied in 'settings.json' in the data folder, double check your bot settings!`))
-	return;
+console.log('---------------------------------------------');
+console.log('TheMonDon\'s DBM Bot Sharder');
+console.log(`Version: ${version}`);
+console.log('You can change the amount of shards by providing \'shard_count=[number]\' (default: auto)');
+console.log('---------------------------------------------');
+
+let totalShards = 'auto';
+
+const args = process.argv
+  .slice(2)
+  .map((val, i) => {
+    const object = {};
+    const [regexForProp, regexForVal] = (() => [/^(.+?)=/, /=(.*)/])();
+    const [prop, value] = (() => [regexForProp.exec(val), regexForVal.exec(val)])();
+    if (!prop) {
+      object[val] = true;
+      return object;
+    } else {
+      object[prop[1]] = value[1];
+      return object;
+    }
+  })
+  .reduce((obj, item) => {
+    const prop = Object.keys(item)[0];
+    obj[prop] = item[prop];
+    return obj;
+  }, {});
+
+if (args && args.shard_count) {
+  console.log(`Command Line Arg: shard_count=${args.shard_count}`);
+  totalShards = Number(args.shard_count);
```

Comment:
Shard Count needs to be an integer. Also, this might be more useful for debugging in case an invalid value is passed by the user
```suggestion
  totalShards = parseInt(args.shard_count, 10);
  console.log(`Command Line Arg: shard_count=${totalShards}`);
```